subsys/bluetooth/shell/csis.c

```diff
@@ -105,20 +105,114 @@ static int cmd_csis_register(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static uint8_t rsi[BT_CSIS_RSI_SIZE];
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_CSIS_DATA_RSI(rsi),
+};
```

Comment:
The CSIP spec states that
> A privacy-enabled Set Member should advertise RSI values only when in the GAP Limited Discoverable mode (see Volume 3, Part C, Section 9.2.3 in [2]).

So perhaps we should reflect that in this adv data as well? i.e. use `BT_LE_AD_LIMITED` instead of `BT_LE_AD_GENERAL` if privacy is enabled.